File: spring_webapp1/src/main/java/com/example/spring_webapp1/domain/printing/IReceiptPrinter.java
package com.example.spring_webapp1.domain.printing;

import java.awt.print.PrinterException;

/**
 * レシートプリンターへの出力を抽象化するポート（依存性逆転の原則）。
 * Infrastructure 層に具体実装を持つ。
 */
public interface IReceiptPrinter {
    void printAndCut(ReceiptDocument document, String printerKeyword) throws PrinterException;
}

File: MauiApp1/Domain/Printing/IReceiptPrinter.cs
namespace MauiApp1.Domain.Printing;

/// <summary>
/// レシートプリンターへの出力を抽象化するポート（依存性逆転の原則）。
/// Infrastructure 層に具体実装を持つ。
/// </summary>
public interface IReceiptPrinter
{
    Task PrintAndCutAsync(
        ReceiptDocument document,
        string? printerName = null,
        CancellationToken cancellationToken = default);
}
